### PR TITLE
MRG: Fix repr

### LINKED ISSUE #2180: FIX: saving RawEDF
Trying to use the save method in a RawEDF object creates the following error ...
screenshot from 2015-06-01 16 34 00

### LINKED ISSUE #2183: BUG: RawArray `_repr_` function fails with `IndexError: list index out of range`
When printing a RawArray object, the `_repr_` function access `self._filenames[0]`, but since it's a RawArray object, it's an empty list, causing a `IndexError: list index out of range`.

### PR DESCRIPTION
**Pull request by Eric89GXL**
Closes #2183.
Closes #2180.

### CODE REVIEW

**Comment by agramfort** [comment]
Lgtm

File: mne/io/edf/edf.py [hunk 1]
```diff
@@ -249,8 +249,10 @@ def _read_segment_file(self, data, idx, offset, fi, start, stop,
         data *= gains.T[sel]
         data += offsets[sel]
 
-        if stim_channel is not None:
-            stim_channel_idx = np.where(sel == stim_channel)[0][0]
+        # only try to read the stim channel if it's not None and it's
+        # actually one of the requested channels
+        if stim_channel is not None and (sel == stim_channel).sum() > 0:
```
Comment (Eric89GXL) [inline mne/io/edf/edf.py]: This is the fix for #2180

**Comment by Eric89GXL** [comment]
CIs are happy, ready for review/merge from my end

**Comment by agramfort** [comment]
thanks !

**Pull request merged**